submissions/AsyaYeromina/Tiny-JS-world/index.js

```diff
@@ -0,0 +1,99 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/AsyaYeromina/a-tiny-JS-world
+   Web app: https://asyayeromina.github.io/a-tiny-JS-world/.
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+const dog = {
+   species: 'Canis',
+   name: 'Jessie',
+   gender: 'female',
+   legs: '4',
+   hands: '0',
+   speech: 'woof-woof!',
+   friends: ['Stepan', 'Ivanka']
+};
+
+const cat = {
+   species: 'Felis catus',
+   name: 'Python',
+   gender: 'male',
+   legs: '4',
+   hands: '0',
+   speech: 'Meow',
+   friends: []
+};
+
+const woman = {
+   species: 'Homo sapiens',
+   name: 'Ivanka',
+   gender: 'female',
+   legs: '2',
+   hands: '2',
+   speech: 'Hello!',
+   friends: ['Stepan', 'Python', 'Jessie']
+};
+
+const man = {
+   species: 'Homo sapiens',
+   name: 'Stepan',
+   gender: 'male',
+   legs: '2',
+   hands: '2',
+   speech: 'Hello!',
+   friends: ['Ivanka', 'Python', 'Jessie']
+};
+
+const catWoman = {
+   species: 'Felis sapiens',
+   name: 'Selina',
+   gender: 'female',
+   legs: '2',
+   hands: '2',
+   speech: cat.speech,
+   friends: ['Stepan', 'Python']
+};
+
+const inhabitants = [cat, dog, woman, man, catWoman];
+const keys = ['species', 'name', 'gender', 'legs', 'hands', 'speech', 'friends'];
+
+const getValues = function(obj, arr) {
+   const stringArr = [];
+   arr.forEach(key => {
+      const value = obj[key];
+      if (typeof value === 'object') {
+         stringArr.push(value.join(', '))
+      } else
+      stringArr.push(value);
+   });
+   return stringArr.join('; ');
```

Comment:
Whenever you have a source array and resulting array of the same dimension, where each source array element translates into something else, then it is a use case for `Array.map`

Consider the following fragment:

```
['propNameA', 'propNameB']
  .map(propName => obj[propName]) // map keys onto values
  .map(value=> Array.isArray(value) ? value.join(", ") : value) // if any value is an array then merge its elements, otherwise leave as it is
  .join("; ");
```
The above `map`s could be merged. In a simple case it is acceptable. In complex cases it is better to have every step have its own distinct and readable operation.

Reply:
Please check the code with map in latest commit